src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -166,28 +394,206 @@ template<class T> bool pointFrustum(const Vector3<T>& point, const Frustum<T>& f
     return true;
 }
 
-template<class T> bool boxFrustum(const Range3D<T>& box, const Frustum<T>& frustum) {
-    for(const Vector4<T>& plane: frustum.planes()) {
-        bool cornerHit = 0;
+template<class T> bool rangeFrustum(const Range3D<T>& box, const Frustum<T>& frustum) {
+    /* Convert to center/extent, avoiding division by 2 and instead comparing
+       to 2*-plane.w() later */
+    const Vector3<T> center = box.min() + box.max();
+    const Vector3<T> extent = box.max() - box.min();
 
-        for(UnsignedByte c = 0; c != 8; ++c) {
-            const Vector3<T> corner = Math::lerp(box.min(), box.max(), Math::BoolVector<3>{c});
+    for(const Vector4<T>& plane: frustum.planes()) {
```

Comment:
Why is this not just calling into the other `rangeFrustum()` overload? The code is the same.

The `box` param should be `range` to be consistent with the declaration above.

Reply:
Because it's comapring with `-T(2)*plane.w()` instead of `-plane.w()`, to avoid the vector/scalar division when converting range to center & extent representation.

Reply:
(As discussed offline: okay, leave it as is.)